opensearch-operator/controllers/opensearchController.go

```diff
@@ -228,6 +239,18 @@ func (r *OpenSearchClusterReconciler) reconcilePhasePending(ctx context.Context)
 }
 
 func (r *OpenSearchClusterReconciler) reconcilePhaseRunning(ctx context.Context) (ctrl.Result, error) {
+	// Update initialized status first
+	if !r.Instance.Status.Initialized {
+		if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			if err := r.Get(ctx, client.ObjectKeyFromObject(r.Instance), r.Instance); err != nil {
+				return err
+			}
+			r.Instance.Status.Initialized = builders.AllMastersReady(ctx, r.Client, r.Instance)
+			return r.Status().Update(ctx, r.Instance)
+		}); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
```

Comment:
@idanl21 this is where I calculate if a cluster is initialized.  It the current status is uninitialized it checks if all masters are ready, and if so sets it to initialized.